src/classes/BDI_CustObjMappingGAUAllocation.cls

```diff
@@ -0,0 +1,102 @@
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Batch Data Import
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Overrides the standard object mapping logic class for GAU allocation field mapping since only one of
+* the Opportunity__c, Payment__c and Campaign__c fields should be populated.
+*/
+
+public with sharing class BDI_CustObjMappingGAUAllocation extends BDI_ObjectMappingLogic {
+
+    public override BDI_ObjectWrapper[] populateObjects(BDI_ObjectWrapper[] objWraps) {
+
+        Allocations_Settings__c settings = UTIL_CustomSettingsFacade.getAllocationsSettings();
+
+        String PAYMENT_FIELDNAME = 'npsp__Payment__c';
+        String OPPORTUNITY_FIELDNAME = 'npsp__Opportunity__c';
+        String CAMPAIGN_FIELDNAME = 'npsp__Campaign__c';
+
+        for (BDI_ObjectWrapper objWrap : objWraps) {
+            Map<String, Schema.DescribeFieldResult> fieldDescribeMap = UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.adjustNPSPNamespace(objWrap.objMapping.Object_API_Name__c));
+
+            objWrap.sObj = Schema.getGlobalDescribe().get(UTIL_Namespace.adjustNPSPNamespace(objWrap.objMapping.Object_API_Name__c)).newSObject();
+
+            if (objWrap.existingSobjectId != null) {
+                objWrap.sObj.put('Id', objWrap.existingSobjectId);
+            }
+
+            Object campaignValue;
+            Object paymentValue;
+            Object opportunityValue;
+
+            for (Data_Import_Field_Mapping__mdt fieldMapping : objWrap.fieldMappings) {
```

Comment:
Similar here with the repeat use of "fieldMapping"

Reply:
updated to fieldMappingCustomMetadata